File: Python/0009_palindrome_number.py
# 2020-08-05 13:00:36

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        remain = x
        total = 0
        while remain != 0:
            digit = remain % 10
            total = total * 10 + digit
            remain //= 10
        return total == x

File: Java/0009_Palindrome_Number.java
// 2020-08-04 22:58:55

class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int remain = x;
        long sum = 0;
        while (remain != 0) {
            int num = remain % 10;
            sum = sum * 10 + num;
            remain /= 10;
        }
        return sum == x;
    }
}
